contracts/vesting/VestingWalletRecovery.sol

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: UNLICENSED
+// See Forta Network License: https://github.com/forta-network/forta-contracts/blob/master/LICENSE.md
+
+pragma solidity ^0.8.9;
+
+import "./VestingWalletV1.sol";
+
+/**
+ * This contract is desigend for recovering the in case the beneficiary was lost.
+ */
+contract VestingWalletRecovery is VestingWalletV1 {
+    event BeneficiaryUpdate(address newBeneficiary);
+
+    function updateBeneficiary(address newBeneficiary) external onlyOwner {
+        _setBeneficiary(newBeneficiary);
+        emit BeneficiaryUpdate(newBeneficiary);
```

Comment:
I'd say this event should be emitted inside of `_setBeneficiary` instead.

Reply:
I wanted to minimize the changes to V1. If the storage was not private I wouldn't have touched it.

Reply:
I see. I was thinking of overriding `_setBeneficiary` in V2:

```
function _setBeneficiary(address newBeneficiary) override ... {
  super._setBeneficiary(newBeneficiary);
  emit BeneficiaryUpdate(newBeneficiary);
}
```

Reply:
feels a bit too much to me. It means having two function instead of one in `VestingWalletRecovery`.


```solidity
contract VestingWalletRecovery is VestingWalletV1 {
    event BeneficiaryUpdate(address newBeneficiary);

    function updateBeneficiary(address newBeneficiary) external onlyOwner {
        _setBeneficiary(newBeneficiary);
    }

    function _setBeneficiary(address newBeneficiary) internal virtual override {
        super._setBeneficiary(newBeneficiary);
        emit BeneficiaryUpdate(newBeneficiary);
    }
}
```

vs

```solidity
contract VestingWalletRecovery is VestingWalletV1 {
    event BeneficiaryUpdate(address newBeneficiary);

    function updateBeneficiary(address newBeneficiary) external onlyOwner {
        _setBeneficiary(newBeneficiary);
        emit BeneficiaryUpdate(newBeneficiary);
    }
}
```

Reply:
I agree with prioritizing fewer code and fewer changes to V1.